pandas/io/json/_json.py

```diff
@@ -716,6 +720,14 @@ def _get_data_from_filepath(self, filepath_or_buffer):
                 errors=self.encoding_errors,
             )
             filepath_or_buffer = self.handles.handle
+        elif (
+            isinstance(filepath_or_buffer, str)
+            and filepath_or_buffer.lower().endswith(
+                (".json",) + tuple(f".json{c}" for c in _extension_to_compression)
+            )
+            and not file_exists(filepath_or_buffer)
+        ):
+            raise FileNotFoundError(f"File {filepath_or_buffer} does not exist")
```

Comment:
After read_json knows that it is a file, I think you can just raise inside get_handle when the file doesn't exist:

```py
if "r" not in mode and is_path and file_exists(handle):
    raise ...
```

Reply:
You mean this, right?
```py
if "r" in mode and is_path and not file_exists(handle):
    raise ...
```

Reply:
Actually `read_json` doesn't even call `get_handle` since it's behind a `file_exists` check:

```py
if (
    not isinstance(filepath_or_buffer, str)
    or is_url(filepath_or_buffer)
    or is_fsspec_url(filepath_or_buffer)
    or file_exists(filepath_or_buffer)
):
    self.handles = get_handle(...)
```

Reply:
I think there were attempts at creating an efficient way of checking whether a str is a legit JSON string. I assume these attempts were unsuccessful - we check the reverse. If that's the case, then moving the check into get_handle, will not work.

Reply:
> If that's the case, then moving the check into get_handle, will not work.

So leave PR as is?

Reply:
> So leave PR as is?

Probably.